spyder/widgets/variableexplorer/tests/test_dataframeeditor.py

```diff
@@ -174,6 +174,15 @@ def test_change_format_with_format_not_starting_with_percent(qtbot, monkeypatch)
     with qtbot.assertNotEmitted(editor.sig_option_changed):
         editor.change_format()
 
+def test_header_encoding():
+    df = DataFrame({"ä": [1],
+                    "ï»¿Unieke_Idcode": ['a'],
+                    1: ['abcd'],
+                    "f": ['31/12/16'],
+                    "": ['']})
+    editor = DataFrameEditor(None)
+    editor.setup_and_check(df)
```

Comment:
Tests always assert that the result of an operation is equal to some expected value. However, here you're not doing that. You're just testing that the DataFrame editor doesn't crash, but not that the header information is correct.

Also, please add a test for issue #2514, as I told you so ;-)